packages/echo-base/src/errors/errorUtils.ts

```diff
@@ -0,0 +1,59 @@
+import { BaseError, findPropertyByName, getAllProperties } from './BaseError';
+import { BackendError, ForbiddenError, NetworkError, NotFoundError, UnauthorizedError } from './NetworkError';
+import { toError } from './toError';
+
+const is = {
+    error: isError,
+    baseError: isBaseError,
+    networkError: isNetworkError,
+    backendError: isBackendError,
+    notFoundError: isNotFoundError,
+    forbiddenError: isForbiddenError,
+    unauthorizedError: isUnauthorizedError
+};
+
+export const errorUtils = {
+    toError,
+    is,
+    findPropertyByName,
+    getAllProperties
+};
+
+// instanceOf does not work if the error was thrown in another module, instead we compare by known properties such as httpStatusCode or errorClass.name
+// also see: https://medium.com/@samjwright/why-is-instanceof-my-object-returning-false-in-typescript-fec74df5c2a8
+
+function isError(err: Error): err is Error {
+    return typeof err === 'object' && !!err && 'message' in err && 'name' in err;
+}
+
+function isBaseError(error: Error): error is BaseError {
+    const hasBeenLoggedValue = findPropertyByName(error, 'hasBeenLogged');
+    return isError(error) && hasBeenLoggedValue !== undefined;
+}
+
+function isNetworkError(error: Error): error is NetworkError {
+    const httpStatusCode = findPropertyByName(error, 'httpStatusCode');
+    //httpStatusCode 0 means there was network problems / missing connection
+    return httpStatusCode >= 0 || error.name === NetworkError.name;
+}
+
+function isBackendError(error: Error): error is BackendError {
+    const httpStatusCode = findPropertyByName(error, 'httpStatusCode');
+    //httpStatusCode > 0 means we got a response from the server
+    return httpStatusCode > 0 || error.name === BackendError.name;
```

Comment:
How about introducing small, well named atomic functions so the comments can be removed:
```
function isServerResponseByHttpCode(httpStatusCode: number): boolean {
    httpStatusCode > 0
}
```

Same could be applied for line 36 + 37:
```
isNetworkErrorByHttpCode(httpStatusCode: number): boolean {
    return httpStatusCode === 0;
}
```

With these `isNetworkError` would look like:
```
function isNetworkError(error: Error): error is NetworkError {
    const httpStatusCode = findPropertyByName(error, 'httpStatusCode');
    return isServerErrorByHttpCode(httpStatusCode) || isNetworkErrorByHttpCode(httpStatusCode) || error.name === NetworkError.name;
}
```